pkg/instrumentation/helper.go

```diff
@@ -123,6 +124,22 @@ func isInstrWithoutContainers(inst instrumentationWithContainers) int {
 	return 0
 }
 
+func instrVolume(volume corev1.Volume, name string, volumeSizeLimit *resource.Quantity) (corev1.Volume, error) {
+	if !reflect.ValueOf(volume).IsZero() && volumeSizeLimit != nil {
+		return volume, fmt.Errorf("both Volume and VolumeSizeLimit cannot be defined simultaneously")
+	} else if !reflect.ValueOf(volume).IsZero() {
+		return volume, nil
+	}
```

Comment:
This check should happen in the instrumentation admission webhook. Here, during Pod injection, we should use the highest priority setting instead of returning an error, in my opinion. In this case, use the volume if it's set, otherwise use the emptydir.

Reply:
Right, I definitely agree. Moved validation logic to webhook + slimmed down helper logic.